### PR TITLE
OCPBUGS-18135: IBMCloud specific: patch out management workload for dataplane component thats needed for bootstrapping

### PR DESCRIPTION
**Pull request by KodieGlosserIBM**
openshift-network-operator runs in the dataplane for IBM Cloud.
4.13 introduced some changes that require worker nodes to be present on the cluster for deployments with this annotation to schedule: target.workload.openshift.io/management.
While this annotation is "preferred" it delays initial provisioning by ~5-15 minutes.

CNO is required for bootstrapping of worker nodes in openshift clusters.

Removing this annotation on the CNO to allow for it to immediately schedule on a node to allow it complete provisioning.

Slack Thread: https://ibm-argonauts.slack.com/archives/C015MKYUVSR/p1691784810280779

Concept doc that introduced this change for 4.13: https://github.com/dhellmann/openshift-enhancements/blob/f2faf86f757269fa1c0bf39023c3ae0911f0d413/enhancements/management-workload-partitioning.md

### CODE REVIEW

**Review by rtheis — COMMENTED** [review]
Is this patching used exclusively for IBM ROKS Toolkit installs or does it apply to HyperShift installs of ROKS too?

**Comment by rtheis** [comment]
@KodieGlosserIBM @csrwng don't we want HyperShift to keep the management workload configuration since the network operator runs in the control plane?

**Comment by KodieGlosserIBM** [comment]
@rtheis sorry I misspoke. Took a deeper look with Hide. HyperShift uses the CNO on the control plane which is deployed by CPO [here](https://github.com/openshift/hypershift/blob/4065bed12eee243b06aa3f75cd6df12820a15e59/control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go#L280)
(this CPO also excludes this network operator we are changing from being deployed in cluster: [here](https://github.com/openshift/hypershift/blob/9b25e7472addf5cc421cbd8eef11b77f26655b40/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L217))

So tl;dr this is only used with our ROKS toolkit. And is ready to go in!

**Review by rtheis — APPROVED** [review]
/lgtm

**Review by trozet — APPROVED** [review]
/approve

**Comment by KodieGlosserIBM** [comment]
Thanks @trozet can we merge this now? And I can cherry pick back to 4.13

**Comment by csrwng** [comment]
/retitle OCPBUGS-18135: IBMCloud specific: patch out management workload for dataplane component thats needed for bootstrapping

**Comment by KodieGlosserIBM** [comment]
@zshi-redhat thanks for taking a look! It looks like the required tests are passing but only critical fixes are allowed?

**Comment by zshi-redhat** [comment]
> @zshi-redhat thanks for taking a look! It looks like the required tests are passing but only critical fixes are allowed?

@KodieGlosserIBM Yes, in order to stabilize the nightly build, we only do critical fix merging at the moment until 4.15 branching. 
can we wait (likely be the end of week) or is it a blocker? I'd like to understand the severity of it if possible.

**Comment by KodieGlosserIBM** [comment]
@zshi-redhat It is a higher priority for us, but Friday seems reasonable. We will need to get this cherry-picked back to 4.13 as well. Thank you!

**Comment by KodieGlosserIBM** [comment]
@zshi-redhat Once we get a new release for 4.14 build we should be able to verify the change!

**Comment by zshi-redhat** [comment]
> @zshi-redhat Once we get a new release for 4.14 build we should be able to verify the change!

@KodieGlosserIBM  cool, we need the 4.14 bug to be in verified state in order for the 4.13 PR to have the valid-bug lablel and merge.

**Comment by zshi-redhat** [comment]
@KodieGlosserIBM  any update on the verification of this change?
If you can leave a comment here that indicating the change is verified in IBMCloud, I will update the jira tracker: https://issues.redhat.com/browse/OCPBUGS-18135, which will then unblock the 4.13 backport: https://github.com/openshift/cluster-network-operator/pull/1994

**Comment by KodieGlosserIBM** [comment]
@zshi-redhat Verified this change works as expected on IBMCloud!

**Comment by zshi-redhat** [comment]
> @zshi-redhat Verified this change works as expected on IBMCloud!

Cool, thanks! I moved the 4.14 bug to verified: https://issues.redhat.com/browse/OCPBUGS-18135
Now I'm requesting reviews on the 4.13 PR.
